[CLOUDCRAFT-766] Component: Network ACL (#24001)
DataDog/documentation: content/en/cloudcraft/components-aws/network-acl.md

---
title: Network ACL
---

## Overview

Use the Network ACL component to visualize network access control lists from your Amazon Web Services architecture.

{{< img src="cloudcraft/components-aws/network-acl/component-network-acl-diagram.png" alt="Screenshot of an isometric Cloudcraft diagram showing the 'Network ACL' AWS component." responsive="true" style="width:60%;">}}

## Toolbar

Use the toolbar to configure and customize the component. The following options are available:

> - **Visibility**. Toggle the visibility of the component on the diagram.

Reviewer: ```suggestion
- **Visibility**: Toggle the visibility of the component on the diagram.
```

> - **Color**. Select a fill color for the body of the component and an accent color for its symbol. You can use the same colors on 2D and 3D views or different colors for each.

Reviewer: ```suggestion
- **Color**: Select a fill color for the body of the component and an accent color for its symbol. You can use the same colors on 2D and 3D views or different colors for each.
```

> - **Name**. Give the NACL a name.

Reviewer: ```suggestion
- **Name**: Give the NACL a name.
```

> - **Shape**. Select a shape for the component.

Reviewer: ```suggestion
- **Shape**: Select a shape for the component.
```

> - **Padding**. Increase or decrease the amount of space inside the component.

Reviewer: ```suggestion
- **Padding**: Increase or decrease the amount of space inside the component.
```

> - **Rules**. Add, edit, or remove inbound and outbound rules for the NACL.

Reviewer: ```suggestion
- **Rules**: Add, edit, or remove inbound and outbound rules for the NACL.
```

Reviewer 2: The rules here are read-only. So actually I would change this to "View inbound and outbound rules for the NACL"

            "ruleNumber": 100,
            "protocol": "-1",
            "portRange": "0",
            "target": "0.0.0.0/0",
            "targetType": "ip",
            "access": "allow"
        },
        {
            "ruleNumber": 32767,
            "protocol": "-1",
            "portRange": "0",
            "target": "0.0.0.0/0",
            "targetType": "ip",
            "access": "deny"
        }
    ],
    "outboundRules": [
        {
            "ruleNumber": 100,
            "protocol": "-1",
            "portRange": "0",
            "target": "0.0.0.0/0",
            "targetType": "ip",
            "access": "allow"
        },
        {
            "ruleNumber": 32767,
            "protocol": "-1",
            "portRange": "0",
            "target": "0.0.0.0/0",
            "targetType": "ip",
            "access": "deny"
        }
    ],
    "color": {
        "isometric": "#5698ff",
        "2d": "#5698ff"
    },
    "link": "https://aws.amazon.com/",
    "locked": true
}
```

- **type: string**: The type of component.
- **id: string**: A unique identifier for the component consisting of the string `acl-` followed by a random 17-character alphanumeric string.

> - **arn: string**: The globally unique identifier for the component within AWS, known as [Amazon Resource Names](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html).

Reviewer: ```suggestion
- **arn: string**: The globally unique identifier for the component within AWS, known as [Amazon Resource Names][2].
```

- **outboundRules: array**: The rules for outgoing traffic for components inside the NACL.
  - **ruleNumber: number**: The rule number for the NACL.
  - **protocol: string**: The protocol for the rule. Accepts either `-1`, meaning "All," or a specific protocol.
  - **portRange: string**: The listening port or port range for the traffic.
  - **target: string**: The destination for the traffic (CIDR range).
  - **targetType: string**: The target type for the rule. Accepts either `ip` or `ipv6`.
  - **access: string**: The type of access for the rule. Accepts either `allow` or `deny`.
- **color: object**: The fill color for the component body.
  - **isometric: string**: The fill color for the component in the 3D view. Must be a hexadecimal color. Defaults to `#ECECED`.
  - **2d: string**: The fill color for the component in the 2D view. Must be a hexadecimal color. Defaults to `#CC2264`.
- **link: uri**: Link the component to another diagram using the `blueprint://ID` format or to an external website using the `https://LINK` format.
- **locked: boolean**: If `true`, changes made to the component using the application are disabled until unlocked.

> [1]: /cloudcraft/api/

Reviewer: ```suggestion
[1]: /cloudcraft/api/
[2]: https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
```
